### PR TITLE
feat(envited.ascs.digital): Error handling asset verification

### LINKED ISSUE #397: Error handling: Asset verification
# Asset uploader: verification steps
We have documented the verification steps in [EVES-003](https://ascs-ev.github.io/EVES/drafts/EVES-003/eves-003.html) and implemented them in the asset uploader.

# Problem
When I started creating test cases (attached) I realized that the message for the user was always the same:
grafik

# Expectation
Each verfication step shall be separated into different functions of a verification module. Each module shall have test cases which confirm functionality. Each error must contain reasonable information for the user to find the issue.
The software module shall be reusable for the executing the verification needed in the indexer requested in #383 

# Some test cases
[asset_missing_data_file.zip](https://github.com/user-attachments/files/18058867/asset_missing_data_file.zip)
[asset_new_files_for_ipfs.zip](https://github.com/user-attachments/files/18058868/asset_new_files_for_ipfs.zip)
[asset_unexpected_file_in_data.zip](https://github.com/user-attachments/files/18058869/asset_unexpected_file_in_data.zip)
[asset_has_external_access_permissions.zip](https://github.com/user-attachments/files/18058870/asset_has_external_access_permissions.zip)
[asset_incorrect_license_contract_ref_did.zip](https://github.com/user-attachments/files/18058871/asset_incorrect_license_contract_ref_did.zip)
[asset_license_missing.zip](https://github.com/user-attachments/files/18058872/asset_license_missing.zip)

### PR DESCRIPTION
**Pull request by jeroenbranje**
# Work done
Add Errors handling for asset verification

- [x] asset_missing_data_file.zip
- [x] asset_missing_readme_file.zip
- [x] asset_new_files_for_ipfs.zip
- [x] asset_unexpected_file_in_data.zip
- [x] asset_has_external_access_permissions.zip
- [x] asset_incorrect_license_contract_ref_did.zip
- [x] asset_license_missing.zip

Closes #397

### CODE REVIEW

File: apps/envited.ascs.digital/common/validator/shacl/shacl.ts [hunk 1]
```diff
@@ -164,12 +178,19 @@ export const validateDomainMetadata = _validateDomainMetadata({
 })
 
 export const _checkIfAllFilesInManifestExists =
-  ({ getShaclDataFromZip }: { getShaclDataFromZip: (file: File, fileName: string) => Promise<string> }) =>
+  ({ getShaclDataFromZip }: { getShaclDataFromZip: (file: File, fileName: string) => Promise<any> }) =>
```
Comment (jdsika) [inline apps/envited.ascs.digital/common/validator/shacl/shacl.ts]: GOT DAMN "ANY" :D

Comment (royscheeren) [inline apps/envited.ascs.digital/common/validator/shacl/shacl.ts]: We need a correct return type here

**Review by royscheeren — CHANGES_REQUESTED** [review]

File: apps/envited.ascs.digital/common/validator/shacl/shacl.ts (hunk 1 above)
Comment (jeroenbranje) [reply apps/envited.ascs.digital/common/validator/shacl/shacl.ts]: I forgot to put it back as it was, when I was fixing the error messages

File: apps/envited.ascs.digital/common/validator/shacl/shacl.utils.ts [hunk 2]
```diff
@@ -56,3 +58,12 @@ export const loadDataset = _loadDataset({
...
 })
+
+export const formatFilesErrorMessage = (errors: { error: string }[]) =>
+  pipe(
+    map(({ error }: { error: string }) => error),
+    join(', '),
+    (x: string) => `${ERRORS.FILES_NOT_FOUND} - ${x}`,
+  )(errors)
+
+export const subtractManifestAndReadMeFiles = flip(subtract)(2)
```
Comment (royscheeren) [inline apps/envited.ascs.digital/common/validator/shacl/shacl.utils.ts]: Can we put the 2 in a constant?

File: apps/envited.ascs.digital/common/validator/shacl/shacl.ts [hunk 3]
```diff
@@ -167,9 +225,26 @@ export const _checkIfAllFilesInManifestExists =
   ({ getShaclDataFromZip }: { getShaclDataFromZip: (file: File, fileName: string) => Promise<string> }) =>
```
Comment (royscheeren) [inline apps/envited.ascs.digital/common/validator/shacl/shacl.ts:225]: Typo: _checkIfAllFilesInManifestExists should be _checkIfAllFilesInManifestExist (no trailing s)

**Review by royscheeren — CHANGES_REQUESTED** [review]

**Review by royscheeren — APPROVED** [review]

**Pull request merged**